File: article-admin-main/src/api/article.ts
import type { Article, ArticleFilter, Category } from '@/types/article.ts'
import { request } from './request'

export interface ArticleListResult {
  items: Article[]
  total: number
}

export function getArticles(params: {
  page: number
  pageSize: number
  filter: ArticleFilter
}) {
  return request<ArticleListResult>({
    url: '/articles/search',
    method: 'post',
    data: {
      page: params.page,
      per_page: params.pageSize,
      keyword: params.filter.keyword,
      section: params.filter.category,
      publish_date_range: {},
    },
  })
}

export function getCategories() {
  return request<[Category]>({ url: '/articles/categories' })
}

File: article-master/app/schemas/article.py
from typing import Optional

from pydantic import BaseModel, Field


class ArticleQuery(BaseModel):
    page: int = Field(default=1, ge=1)
    per_page: int = Field(default=10, ge=1, le=100)
    keyword: Optional[str] = None
    publish_date_range: Optional[dict] = None
    section: Optional[str] = None
    category: Optional[str] = None
    website: Optional[str] = None
